extension-androidauto/src/main/java/com/mapbox/maps/extension/androidauto/MapboxCarMap.kt

```diff
@@ -71,11 +72,29 @@ class MapboxCarMap {
     carContext: CarContext,
     mapInitOptions: MapInitOptions,
   ) = apply {
+    val surfaceCallback = prepareSurfaceCallback(carContext, mapInitOptions)
+    carContext.getCarService(AppManager::class.java).setSurfaceCallback(surfaceCallback)
+  }
+
+  /**
+   * Instead of using [setup], this function allows you to create your own [SurfaceCallback] and
+   * forward the calls to the returned [SurfaceCallback]. This makes it possible for you to adopt
+   * new api versions or intercept the calls and continue to use the [MapboxCarMap] as designed.
+   *
+   * This may be a temporary solution, while androidx.car.app:app:1.3.0 is rolling out
+   * [SurfaceCallback.onClick]. If there is no use for this function in the future, it will be
+   * removed.
+   */
+  @MapboxExperimental
+  fun prepareSurfaceCallback(
```

Comment:
so it's either `MapboxCarMap.setup()` or `prepareSurfaceCallback`? shall we also link to the example on how to use it?

Also could we think of a more intuitive name for this function? currently from the name itself it's not very clear that the surface callback should be set to the CarSurface using 
```
carContext.getCarService(AppManager::class.java)
          .setSurfaceCallback()
```

Reply:
I guess we should also warn the user don't call setup again, since the surfacecallback might be overwritten again..

Reply:
https://github.com/mapbox/mapbox-maps-android/pull/1683#discussion_r977386997

Yeah naming is hard but ultimately am happy with `prepareSurfaceCallback`. Good thing is that `setup` uses the `prepare` function.

I'm not sure what happens if you call `AppManager.setSurfaceCallback` multiple times. If the jetpack actually dispatches callbacks to the new surface. Maybe we should add the ability to change the surface callback at runtime instead of warning users 🤔 

That topic is a little out of scope here, so am going to follow up on it

Reply:
I just tested this just means the new surface callback is ignored.

We don't really have control over how this function is used. 
```
carContext.getCarService(AppManager::class.java).setSurfaceCallback
```

The behavior of that function seems like it could potentially change, considering the first call is accepted and all other calls are ignored. Warnings and errors will not really help anything IMO

Reply:
Well, I'm going to keep following up on this. My dev testing had bugs in it

```
2022-09-23 09:09:27.342 CarMapSurfaceOwner]: onSurfaceAvailable [Surface(name=null)/@0xc509d26, 800x400, dpi: 160]
2022-09-23 09:09:27.377 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 112 - 716, 400) [top: 112.0, left: 0.0, bottom: 0.0, right: 84.0]
2022-09-23 09:09:27.395 CarMapSurfaceOwner]: onVisibleAreaChanged visibleArea:Rect(0, 112 - 716, 400) surface:[Surface(name=null)/@0xc509d26, 800x400, dpi: 160]
2022-09-23 09:09:27.395 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 112 - 716, 400) [top: 112.0, left: 0.0, bottom: 0.0, right: 84.0]
2022-09-23 09:09:27.395 CarMapSurfaceOwner]: onStableAreaChanged stableArea:Rect(0, 112 - 716, 400)
2022-09-23 09:09:27.395 CarMapSurfaceOwner]: notifyStableAreaChanged Rect(0, 112 - 716, 400) [top: 112.0, left: 0.0, bottom: 0.0, right: 84.0]
495 CarMapSurfaceOwner]: onVisibleAreaChanged visibleArea:Rect(0, 32 - 800, 400) surface:[Surface(name=null)/@0xc509d26, 800x400, dpi: 160]

// call setSurfaceCallback
2022-09-23 09:09:38.495 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 32 - 800, 400) [top: 32.0, left: 0.0, bottom: 0.0, right: 0.0]
2022-09-23 09:09:38.501 CarMapSurfaceOwner]: onSurfaceAvailable [Surface(name=null)/@0x4d25325, 800x400, dpi: 160]
2022-09-23 09:09:38.567 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 32 - 800, 400) [top: 32.0, left: 0.0, bottom: 0.0, right: 0.0]
2022-09-23 09:09:38.584 CarMapSurfaceOwner]: onVisibleAreaChanged visibleArea:Rect(0, 32 - 800, 400) surface:[Surface(name=null)/@0x4d25325, 800x400, dpi: 160]
2022-09-23 09:09:38.585 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 32 - 800, 400) [top: 32.0, left: 0.0, bottom: 0.0, right: 0.0]
2022-09-23 09:09:38.585 CarMapSurfaceOwner]: onStableAreaChanged stableArea:Rect(0, 112 - 716, 400)
2022-09-23 09:09:38.586 CarMapSurfaceOwner]: notifyStableAreaChanged Rect(0, 112 - 716, 400) [top: 112.0, left: 0.0, bottom: 0.0, right: 84.0]
2022-09-23 09:09:38.587 CarMapSurfaceOwner]: onSurfaceAvailable [Surface(name=null)/@0xbe5447b, 800x400, dpi: 160]
2022-09-23 09:09:38.637 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 32 - 800, 400) [top: 32.0, left: 0.0, bottom: 0.0, right: 0.0]
2022-09-23 09:09:38.639 CarMapSurfaceOwner]: onVisibleAreaChanged visibleArea:Rect(0, 32 - 800, 400) surface:[Surface(name=null)/@0xbe5447b, 800x400, dpi: 160]
2022-09-23 09:09:38.639 CarMapSurfaceOwner]: notifyVisibleAreaChanged Rect(0, 32 - 800, 400) [top: 32.0, left: 0.0, bottom: 0.0, right: 0.0]
2022-09-23 09:09:38.640 CarMapSurfaceOwner]: onStableAreaChanged stableArea:Rect(0, 112 - 716, 400)
2022-09-23 09:09:38.640 CarMapSurfaceOwner]: notifyStableAreaChanged Rect(0, 112 - 716, 400) [top: 112.0, left: 0.0, bottom: 0.0, right: 84.0]
```

Reply:
Yeah nice it works!

Reply:
There is a small bug to ensure the old surface is destroyed.

But will use https://github.com/mapbox/mapbox-maps-android/pull/1682 for testing

Reply:
Also just to link more things from the investigation. This test case is the reason it all works

https://github.com/mapbox/mapbox-maps-android/blob/3f51e3fafe03fb64fffa8b0c6a44958848961be1/extension-androidauto/src/test/java/com/mapbox/maps/extension/androidauto/CarMapSurfaceOwnerTest.kt#L220-L264

Essentially, when a new surface becomes available. All observers are notified that the old one is detached and then a new one is attached. It does cause the map to flicker, so if you did this setting in real life I would add the setting in a setting screen, or just keep it enabled